recipes/opencv/2.x/conanfile.py

```diff
@@ -0,0 +1,262 @@
+                major, minor = version_tokens[0], version_tokens[1]
+                suffix = "%s_%s" % (major, minor)
+                names = ["%s-%s" % (name, suffix),
+                         "%s-%s_s" % (name, suffix),
+                         "%s-%s_d" % (name, suffix),
+                         "%s-%s_s_d" % (name, suffix),
+                         "%s" % name,
+                         "%s_s" % name,
+                         "%s_d" % name,
+                         "%s_s_d" % name]
+                return " ".join(names)
+
+            for lib in ["Half", "Iex", "Imath", "IlmImf", "IlmThread"]:
+                tools.replace_in_file(find_openexr, "NAMES %s" % lib, "NAMES %s" % openexr_library_names(lib))
+
+            tools.replace_in_file(os.path.join(self._source_subfolder, "CMakeLists.txt"),
+                                "project(OpenCV CXX C)", "project(OpenCV CXX C)\nset(CMAKE_CXX_STANDARD 11)")
+
+        for cascade in ["lbpcascades", "haarcascades"]:
+            tools.replace_in_file(os.path.join(self._source_subfolder, "data", "CMakeLists.txt"),
+                                  "share/OpenCV/%s" % cascade, "res/%s" % cascade)
+
+        tools.replace_in_file(os.path.join(self._source_subfolder, "CMakeLists.txt"), "staticlib", "lib")
+        tools.replace_in_file(os.path.join(self._source_subfolder, "CMakeLists.txt"), "ANDROID OR NOT UNIX", "FALSE")
+        tools.replace_in_file(os.path.join(self._source_subfolder, "CMakeLists.txt"), "${OpenCV_ARCH}/${OpenCV_RUNTIME}/", "")
+        tools.replace_in_file(os.path.join(self._source_subfolder, "modules", "highgui", "CMakeLists.txt"), "JASPER_", "Jasper_")
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.definitions["BUILD_EXAMPLES"] = False
+        self._cmake.definitions["BUILD_DOCS"] = False
+        self._cmake.definitions["BUILD_TESTS"] = False
+        self._cmake.definitions["BUILD_PACKAGE"] = False
+        self._cmake.definitions["BUILD_PERF_TESTS"] = False
+        self._cmake.definitions["BUILD_opencv_apps"] = False
+        self._cmake.definitions["BUILD_opencv_java"] = False
+        self._cmake.definitions["BUILD_ZLIB"] = False
+        self._cmake.definitions["BUILD_JPEG"] = False
+        self._cmake.definitions["BUILD_PNG"] = False
+        self._cmake.definitions["BUILD_TIFF"] = False
+        self._cmake.definitions["BUILD_JASPER"] = False
+        self._cmake.definitions["BUILD_OPENEXR"] = False
+
+        self._cmake.definitions["WITH_CUFFT"] = False
+        self._cmake.definitions["WITH_CUBLAS"] = False
+        self._cmake.definitions["WITH_NVCUVID"] = False
+        self._cmake.definitions["WITH_FFMPEG"] = False
+        self._cmake.definitions["WITH_GSTREAMER"] = False
+        self._cmake.definitions["WITH_OPENCL"] = False
+        self._cmake.definitions["WITH_CUDA"] = False
+        self._cmake.definitions["WITH_ZLIB"] = True
```

Comment:
Just asking, I'm no opencv user.
Would adding a `with_zlib` option make sense?

Reply:
my mistake, this option is invalid. zlib is mandatory for opencv_core, only BUILD_ZLIB is a valid option.

Reply:
Indeed.
@madebr, check those options: https://github.com/opencv/opencv/blob/51cfa519249c89d24035b3f96315ad997deb300f/CMakeLists.txt#L137

Reply:
Indeed, thanks for checking this!